books/xdoc/fancy/xdoc.js

```diff
@@ -1051,184 +1051,131 @@ document.addEventListener("DOMContentLoaded", () => {
 });
 
 $(document).ready(function()
-{
-    // Load the xindex content.
-    const xindexLoad = loadJS("./xindex.js").then(() => {
-        xindexObj.loadFromXindex(xindex);
-        xindex_loaded = true;
-    });
-    const xsltLoad = loadJS("./render.js").then(() => {
-        const xsltDecoded = atob(xslt_base64);
-        xdocRenderer.init(xsltDecoded);
-    });
-    // Ensure that both the index and the XSL template are loaded before
-    // continuing.
-    Promise.all([xindexLoad, xsltLoad]).then(_ => {
-        onIndexLoaded();
-    });
-    maybePowertip(".toolbutton", {placement: 'se'});
-    maybePowertip(".rtoolbutton", {placement: 'sw'});
-});
+		  {
+		      // Load the xindex content.
+		      const xindexLoad = loadJS("./xindex.js").then(() => {
+			  xindexObj.loadFromXindex(xindex);
+			  xindex_loaded = true;
+		      });
+		      const xsltLoad = loadJS("./render.js").then(() => {
+			  const xsltDecoded = atob(xslt_base64);
+			  xdocRenderer.init(xsltDecoded);
+		      });
+		      // Ensure that both the index and the XSL template are loaded before
+		      // continuing.
+		      Promise.all([xindexLoad, xsltLoad]).then(_ => {
+			  onIndexLoaded();
+		      });
+		      maybePowertip(".toolbutton", {placement: 'se'});
+		      maybePowertip(".rtoolbutton", {placement: 'sw'});
+		  });
 
 function jumpRender(datum) {
     var key = datum["value"];
     var ret = "";
     ret += "<p>";
-//    ret += xindexObj.topicUid(key) + " &mdash; "; // nice for debugging
+    //    ret += xindexObj.topicUid(key) + " &mdash; "; // nice for debugging
     ret += "<b class=\"sf\">" + xindexObj.topicName(key) + "</b>";
     var shortmsg = topicShortPlaintext(key);
     if (shortmsg != "") {
-	ret += " &mdash; " + shortmsg;
+        ret += " &mdash; " + shortmsg;
     }
     ret += "<br/><tt>" + key + "</tt></p>";
     return ret;
 }
 
 function jumpInit() {
-
     var ta_data = [];
-    var keys = xindexObj.allKeys();
+    var keys = Array.from(xindexObj.allKeys());
```

Comment:
Is the `Array.from` necessary? I think `allKeys` returns an iterator. If it works without `Arrays.from`, probably best not to change it.